ptero_workflow/implementation/backend.py

```diff
@@ -216,6 +216,47 @@ def get_workflow_executions(self, workflow_id, since=None):
         else:
             return [], None
 
+    def get_limited_workflow_executions(self, **kwargs):
+        model_class = models.Execution
+        return self._get_limited_reports(model_class=model_class, **kwargs)
+
+    def _get_limited_reports(self, model_class, workflow_id,
+            limit, since=None):
+        query = self.session.query(model_class)
+
+        if since is not None:
+            query = query.filter(model_class.workflow_id == workflow_id,
+                            model_class.timestamp >= since)
+        else:
+            query = query.filter(model_class.workflow_id == workflow_id)
+
+        query = query.order_by(model_class.timestamp).limit(limit + 1)
+        instances = query.all()
+
+        if instances:
+            if len(instances) == limit + 1:
+                # ensure that we return exactly the right number of results
+                next_execution = instances.pop()
+            else:
+                next_execution = instances[-1]
+
+            timestamp = next_execution.timestamp
+            reports = [e.as_dict_for_limited_report()
+                    for e in instances]
+            num_remaining = self.session.query(model_class).\
+                    filter(model_class.id > instances[-1].id,
```

Comment:
I decided to use ID here instead of timestamp because of the potential degeneracy in the timestamp field (duplicate values).  It seems less bad to me than putting it in the API since, we are a lot closer to the implementation here and therefore it isn't as much of a coincidence as a choice.  All that being said, I don't know what will happen when/if postgresql starts recycling id values.

This could definitely be re-written to not use the ID field though, so I'll fix it.